### PR TITLE
syncserver service: follow-up of #28189

### PR DESCRIPTION
**Pull request by Nadrieril**
###### Motivation for this change

https://github.com/NixOS/nixpkgs/pull/28189 still required some changes but I didn't get time to do them all before it was merged.

###### Things done

- fix the permission problem pointed out by @Mic92 
- remove `user` and `group` options
- add myself as maintainer

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `build-use-sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] Linux
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by Nadrieril** [comment]
I see you added the "port to stable" label. Could you tell me what I should do to make this happen ?

File: nixos/modules/services/networking/firefox/sync-server.nix [hunk 1]
```diff
@@ -70,18 +72,6 @@ in
...
-        type = types.str;
-        default = "syncserver";
-        description = "User account under which syncserver runs.";
-      };
-
-      group = mkOption {
-        type = types.str;
-        default = "syncserver";
-        description = "Group account under which syncserver runs.";
-      };
```
Comment (Mic92) [inline nixos/modules/services/networking/firefox/sync-server.nix]: Can you add this to `nixos/modules/rename.nix` using `mkRemovedOptionModule`?

Comment (Nadrieril) [reply nixos/modules/services/networking/firefox/sync-server.nix]: Done :)

**Comment by joachifm** [comment]
Thank you.

If you want to get this into stable as well, please cherry-pick (with `-x`) & submit a PR against the release branch.

**Pull request merged**